ztools/ztls/handshake_messages.go

```diff
@@ -576,6 +631,13 @@ func (m *serverHelloMsg) marshal() []byte {
 		extensionsLength += 1
 		numExtensions++
 	}
+	if alpnLen := len(m.alpnProtocol); alpnLen > 0 {
+		if alpnLen >= 256 {
+			panic("invalid ALPN protocol")
```

Comment:
I think it is appropriate to panic() here. This panic is not triggered by server, but by client, when he's doing something very wrong: you'd want to know about misconfiguration.